src/main/kotlin/no/nav/syfo/persistering/db/PersisterDialogmeldingQueries.kt

```diff
@@ -109,8 +116,164 @@ fun Connection.erDialogmeldingOpplysningerLagret(dialogmeldingid: String) =
+            if (updated != 1) {
+                throw SQLException("Expected a single row to be updated, got update count $updated")
+            }
+        }
+        connection.commit()
+    }
+}
+
+fun DatabaseInterface.erDialogmeldingOpplysningerSendtKafka(dialogmeldingid: String) =
+    connection.use { connection ->
+        connection.prepareStatement(
+            """
+                SELECT dialogmelding_published
+                FROM DIALOGMELDINGOPPLYSNINGER
+                WHERE id=?;
+                """
+        ).use {
+            it.setString(1, dialogmeldingid)
+            val list = it.executeQuery().toList { getTimestamp("dialogmelding_published") }
+            list.isNotEmpty() && list.firstOrNull() != null
+        }
+    }
+
+fun DatabaseInterface.lagreSendtKafka(dialogmeldingid: String) {
+    connection.use { connection ->
+        connection.prepareStatement(
+            """
+                UPDATE DIALOGMELDINGOPPLYSNINGER 
+                SET dialogmelding_published=?
+                WHERE ID=?;
+                """
+        ).use {
+            it.setTimestamp(1, Timestamp.valueOf(LocalDateTime.now()))
+            it.setString(2, dialogmeldingid)
+            val updated = it.executeUpdate()
+            if (updated != 1) {
+                throw SQLException("Expected a single row to be updated, got update count $updated")
+            }
+        }
+        connection.commit()
+    }
+}
+
+fun DatabaseInterface.erDialogmeldingOpplysningerSendtApprec(dialogmeldingid: String) =
+    connection.use { connection ->
+        connection.prepareStatement(
+            """
+                SELECT apprec
+                FROM DIALOGMELDINGOPPLYSNINGER
+                WHERE id=?;
+                """
+        ).use {
+            it.setString(1, dialogmeldingid)
+            val list = it.executeQuery().toList { getTimestamp("apprec") }
+            list.isNotEmpty() && list.firstOrNull() != null
+        }
+    }
+
+fun DatabaseInterface.lagreSendtApprec(dialogmeldingid: String) {
+    connection.use { connection ->
+        connection.prepareStatement(
+            """
+                UPDATE DIALOGMELDINGOPPLYSNINGER 
+                SET apprec=?
+                WHERE ID=?;
+                """
+        ).use {
+            it.setTimestamp(1, Timestamp.valueOf(LocalDateTime.now()))
+            it.setString(2, dialogmeldingid)
+            val updated = it.executeUpdate()
+            if (updated != 1) {
+                throw SQLException("Expected a single row to be updated, got update count $updated")
+            }
+        }
+        connection.commit()
+    }
+}
+
+fun DatabaseInterface.hentIkkeFullforteDialogmeldinger() =
+    connection.use { connection ->
+        connection.prepareStatement(
+            """
+                SELECT id, fellesformat
+                FROM dialogmeldingopplysninger
+                WHERE apprec IS NULL AND mottatt_tidspunkt < (NOW() - INTERVAL '2 minutes')
```

Comment:
Krever at meldingen er 2 minutter gammel for at den skal bli forsøkt reprosessert: Tanken er at det ikke skal bli samtidighetskonflikter ved av cronjob'en begynner å reprosessere en dialogmelding som ikke er ferdig ennå.

Reply:
Den må muligens øke litt nå som Joarkkallet har satt socket timeout til 1 minutt, og den forsøker 3 ganger?

Reply:
Økt til 10 minutter